torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1507,10 +1507,13 @@ int listNe<at::Tensor>(Stack& stack) {
   return 0;
 }
 
+template <class T>
 Operation listList(const Node* node) {
   return [=](Stack& stack) {
-    // Intentional no-op, needed to match Python semantics for list(iterable),
-    // but in JIT these will already be lists
+    c10::ListPtr<T> a = c10::make_list<T>();
```

Comment:
Is there no default constructor? `pop` will probably overwrite whatever there is in this value, so there's no need to initialize it beforehand I think.

Reply:
Everywhere else Sebastian rewrote it in this manner, so I assume it's the right way to do it.

Reply:
I just skimmed over the implementation of `pop` and it will simply overwrite whatever is in that value, so allocating a fresh list seems very wasteful. cc @smessmer

Reply:
Use `c10::List<T> a = pop(stack)` if you want to avoid the allocation.

Note: `c10::ListPtr` was renamed to `c10::List` and will soon have a default constructor, but the default constructor will still do an allocation.

Reply:
Alright done.